frontends/web/src/routes/exchange/btcdirect.tsx

```diff
@@ -14,35 +14,135 @@
+  const [iframeLoaded, setIframeLoaded] = useState(false);
+  const [height, setHeight] = useState(0);
+
+  const config = useLoad(getConfig);
+  const moonpay = useLoad(getMoonpayBuyInfo(code)); // TODO: getBTCDirectBuyInfo(code)
 
-  const openBTCDirect = () => {
-    open(getBTCDirectLink());
-    navigate(-1);
+  const account = findAccount(accounts, code);
+  const ref = createRef<HTMLDivElement>();
+  let resizeTimerID: any;
+
+  useEffect(() => {
+    if (config) {
+      setAgreedTerms(config.frontend.skipBTCDirectDisclaimer);
+    }
+  }, [config]);
+
+  useEffect(() => {
+    onResize();
+    window.addEventListener('resize', onResize);
+
+    return () => window.removeEventListener('resize', onResize);
+  });
+
+  const onResize = () => {
+    if (resizeTimerID) {
+      clearTimeout(resizeTimerID);
+    }
+    resizeTimerID = setTimeout(() => {
+      if (!ref.current) {
+        return;
+      }
+      setHeight(ref.current.offsetHeight);
+    }, 200);
   };
 
+  if (!account || !config) {
+    return null;
+  }
+
+  const hasOnlyBTCAccounts = accounts.every(({ coinCode }) => isBitcoinOnly(coinCode));
+  const translationContext = hasOnlyBTCAccounts ? 'bitcoin' : 'crypto';
+
   return (
     <div className="contentWithGuide">
       <div className="container">
-        <div className={style.header}>
-          <Header title={<h2>{t('buy.exchange.infoContent.btcdirect.disclaimer.title')}</h2>} />
-        </div>
-        <div className={style.container}>
-          <BTCDirectTerms
-            onContinue={() => openBTCDirect()}
-          />
+        <div className="innerContainer">
+          <div className={style.header}>
+            <Header title={
+              <h2>
+                {t('generic.buy', { context: translationContext })}
+              </h2>
+            } />
+          </div>
+          <div ref={ref} className={style.container}>
+            { !agreedTerms ? (
+              <BTCDirectTerms
+                account={account}
+                onAgreedTerms={() => setAgreedTerms(true)}
+              />
+            ) : (
+              <div style={{ height }}>
+                <UseDisableBackButton />
+                {!iframeLoaded && <Spinner text={t('loading')} />}
+                { address && moonpay && (
+                  <iframe
+                    onLoad={() => {
+                      setIframeLoaded(true);
+                      onResize();
+                    }}
+                    title="BTC Direct"
+                    width="100%"
+                    height={height}
+                    frameBorder="0"
+                    className={`${style.iframe} ${!iframeLoaded ? style.hide : ''}`}
+                    allow="camera"
+                    data-locale={getLocaleWithRegion(i18n.language)}
+                    data-theme={isDarkMode ? 'dark' : 'light'}
+                    data-base-currency={account.coinUnit}
+                    data-quote-currency={'EUR'}
```

Comment:
TODO: pass current selected conversion currency.

Reply:
actually "At this moment the only fiat currency available is Euro (EUR)."
https://developer.btcdirect.eu/widget/supported-fiat-currencies.html